### PR TITLE
KT-18518

### PR DESCRIPTION
**Pull request by Dimach**
Rewritten polyadic expression converter for correct type handling, KT-18518 fixed.

### CODE REVIEW

File: j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt [hunk 1]
```diff
@@ -241,7 +241,8 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
         }
         val name = if (primitiveType != null) {
             "javaPrimitiveType"
-        } else {
+        }
+        else {
```
Comment (lynnfield) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: It is wrong according to Java Coding Conventions

File: j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt [hunk 2]
```diff
@@ -867,7 +912,8 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
                 if (parameters.size == 1 && !isKotlinFunctionType) {
                     // for lambdas all parameters with types should be present
                     emptyList()
-                } else {
+                }
+                else {
```
Comment (piotrek1543) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: Please restore previous code formatting - actual `if-else` statements and spaces are against accepted conceptions.

Comment (yole) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: @piotrek1543 Please let the Kotlin team decide on the coding conventions used in the Kotlin project. Thanks.

**Comment by goodwinnk** [comment]
@Dimach Thank you for your time and effort you put to this pull request, and I'm sorry that we can't merge it now as it does the fix in the previous version of the Java to Kotlin converter that is now unsupported.

There were a number of design issues in the old Java to Kotlin converter that forced us to make a major rewrite of the whole module. The new converter can be found in the nj2k directory in the repository and already enabled by default in 1.3.50-eap2.

When we had started this rewrite we didn't know the final list of issues that are going to be fixed or unresolved at the end.

KT-18518 is still reproducible in the new Java to Kotlin converter that is especially sad because code from this pull request can't be reused directly to fix the issue. Unfortunately fixing should be started from the ground.

I'm sorry for this situation and really appreciate your effort on this pull request.

**Pull request closed without merging**